knimin/handlers/ag_pulldown.py

```diff
@@ -41,8 +41,10 @@ def post(self):
         metadata, failures = db.pulldown(barcodes, blanks)
 
         meta_zip = InMemoryZip()
-        failtext = ("The following barcodes were not retrieved for any "
-                    "survey:\n%s" % "\n".join(failures))
+        failed = '\n'.join(['\t'.join(x) for x in
+                            db.explain_pulldown_failures(failures)])
```

Comment:
why is a separate method call needed, can't this just be reported during the `pulldown` on like 41?